### PR TITLE
Fix parsing quoted strings (#20)

### LINKED ISSUE #20: Error when parsing single-quoted URL strings
There is a bug with single-quoted URLs.

Input:
```
'https://example.org''
```

Expected:
```
https://example.org
```

Actually extracted link:
```
https://example.org''
```

Downstream issue in lychee: https://github.com/lycheeverse/lychee/issues/230

### PR DESCRIPTION
**Pull request by mre**
Fixes #20 

* [x] Add test for single-quoted string
* [x] Fix parsing problem

### CODE REVIEW

File: src/url.rs [hunk 1]
```diff
@@ -2,6 +2,8 @@ use std::ops::Range;
 
 use crate::scanner::Scanner;
 
+const QUOTES: &[char] = &['\'', '\"'];
```
Comment (mre) [inline src/url.rs:5]: We can add more quotes here or even use a crate for that. Wanted to keep it simple for now to make refactoring easier in the future.

File: src/url.rs [hunk 2]
```diff
@@ -28,16 +30,23 @@ impl Scanner for UrlScanner {
 
 impl UrlScanner {
     // See "scheme" in RFC 3986
-    fn find_start(&self, s: &str) -> Option<usize> {
+    fn find_start(&self, s: &str) -> (Option<usize>, Option<char>) {
```
Comment (mre) [inline src/url.rs:33]: I used a tuple type as a return value here.
Alternatively we could introduce a `Marker` struct:

```rust
struct Marker {
  first_char: usize,
  quote: Option<char>,
}
```

And then return `Option<Marker>` here instead.

**Comment by mre** [comment]
Any changes needed before we can merge this?

**Comment by robinst** [comment]
Hey! Thanks for pinging again. LGTM!

**Pull request merged**